runtime/System/Console/Hawk/Runtime.hs

```diff
@@ -74,17 +85,23 @@ splitAtSeparator (Delimiter d) = Search.split d
 
 
 outputRows :: Rows a => OutputSpec -> a -> IO ()
-outputRows (OutputSpec _ spec) x = ignoringBrokenPipe $ do
+outputRows (OutputSpec out spec) x = ignoringBrokenPipe $ do
     let s = join' (toRows x)
-    B.putStr s
-    hFlush stdout
+    case out of
+        UseStdout           -> do B.putStr s; hFlush stdout
+        OutputFile f ""     -> do B.writeFile f s; hFlush stdout
+        OutputFile f backup -> do
+            copyFile f backup
+            B.writeFile f s;
+            hFlush stdout
```

Comment:
Could we move the backup (and eventually the "copy over original on success") logic out of the runtime? I would prefer to keep the runtime as small as possible.

Reply:
On second thought, since `OutputSpec` is defined in the runtime, I guess it does make more sense for that logic to be in the runtime.